### PR TITLE
Boost deprecation warning

### PR DESCRIPTION
**Pull request by d-b-w**
#### What does this implement/fix? Explain your changes.
Addresses a build warning on mac from a deprecated boost header.
RDKit doesn't directly use this deprecated header - it's used in the
depths of boost::random.

I didn't see a way to choose an alternate implementation at a
higher level, so this just hides the deprecation warning.

#### Any other comments?
Mostly posting so that I can see whether the appveyer build
failure in https://github.com/rdkit/rdkit/pull/2351 is related
to my change.

### CODE REVIEW

File: Code/RDGeneral/utils.h [hunk 1]
```diff
@@ -15,7 +15,9 @@
 #include "types.h"
 #include <RDGeneral/Invariant.h>
 #include <RDGeneral/BoostStartInclude.h>
+#define BOOST_ALLOW_DEPRECATED_HEADERS
```
Comment (bp-kelley) [inline Code/RDGeneral/utils.h]: Why not put this in BoostStartInclude/BoostEndInclude?  This is where we hide most of the other warnings so they will be used in all of the files that import boost headers.

Comment (d-b-w) [reply Code/RDGeneral/utils.h]: aha! because I wasn't thinking of that. Updating.

Comment (d-b-w) [reply Code/RDGeneral/utils.h]: actually.... Do we want that?

I would think that we'd want to know if we included any boost headers that were deprecated. The current location hides a single known issue. But if, say, in the future `boost::graph::adjacency_list` were deprecated (to pick a crazy example), I would think we'd want to know as soon as possible.

**Comment by bp-kelley** [comment]
I think you are probably right here.  If it seems endemic, we can move that into the squash warning headers.

**Pull request merged**